adoc/chapters/programming_interface.adoc

```diff
@@ -5961,7 +5962,7 @@ property::image::use_mutex
       application via a [code]#std::mutex# provided to the
       property. The [code]#std::mutex# is locked by
       the runtime whenever the data is in use and unlocked
-      otherwise. Data is synchronized with [code]#hostData#, when
+      otherwise. Data is copied back to [code]#hostData# when
```

Comment:
The way this is worded suggests that data will always be copied back once the mutex is unlocked, but if the buffer would ordinarily wait on commands accessing that data on destruction it will also coordinate, and if the buffer would ordinarily not wait, then there will be no copy back.

Reply:
I don't think I understand how to fix this, primarily because I still don't think I understand how all the buffer destruction rules work.  Could you suggest alternative wording to fix this?

Reply:
I think we could rewrite that last sentence to read:

> The contents of `hostData` are guaranteed to reflect contents of the image when the `std::mutex` is unlocked by the runtime.

Note that this paragraph describes the `use_mutex` property for images.  I think there is a similar paragraph earlier in the spec that describes the `use_mutex` property for buffers.  That should also be changed.

Reply:
I've made both changes in d341f85357d1c102216bee74cf307dc13fdfe287.

Reply:
I think this wording makes sense, though I think we should also clarify that this only occurs if there is a host pointer to copy to.

Reply:
Is there any reason to use `use_mutex` if there isn't a host pointer?  I've not used it, but my understanding from reading the spec was that it's only useful to synchronize access to `hostData`.

Reply:
Not really, though technically it is possible to create a buffer with `use_mutex` and without a host pointer and mutex would be on any host allocation for the memory, you could also use `set_final_data` to set a host pointer.

Saying that I think it's fair to assume that it's implicit that the contents of `hostData` is only reflected if a host pointer is provided, so I think the wording is fine as it is.